book/src/exec.md

```diff
@@ -0,0 +1,169 @@
+
+Intents specify ephemeral interests of users. The step of intent creation consists of two parts:
+1. **Specify an intent**. `IntentVP` encodes the user interests in the [VP](./validity-predicates.md) form. As in a valid transaction all VPs must evaluate to `True`, the intent is guaranteed to be satisfied
+2. **Create the initial *partial transaction*** 
+
+### Partial transaction
+
+Two of the main requirements for an Anoma transaction to be valid are: 
+1. VPs of all involved in the transaction parties must be satisfied
+2. The transaction must be **balanced** , i.e the value spent in the transaction minus the value output equals to some predefined value `v`, the **balancing value**
+
+We call a **partial transaction** a state transition where the first requirement is satisfied (all VPs evaluate to True), but the second one isn't (the state transition is unbalanced). Such a state transition isn't a valid transaction and cannot be published on the blockchain, 
+but can be combined with other partial transactions in order to build a valid transaction. 
+
+![img.png](img/exec_partial_tx.png)
+
+Partial transactions are immutable objects, and once they are created, they cannot be changed. 
+In principle, any party can create a partial transaction, if they can prove that the VPs of the parties involved in the partial transaction are satisfied. 
+
+**Initial partial transactions** are the partial transactions that users create at the beginning, spending the notes that will be given to the counterparty once the intents are matched.
+
+**Note**: For simplicity we assume that intents imply both sending notes and receving notes
+
+As a part of the initial partial transaction, users also output **intent token notes**. 
+### Intent tokens
+
+**Intent tokens** are a mechanism that makes sure that the partial transaction is *unbalanced* until the corresponding intent is satisfied, making it impossible to publish the transaction.
+
+Intent token notes *cannot be spent*, only created. Instead of spending a note of such type, intent notes with *negative value* can be created.
+
+How intent tokens are used:
+1. Spending notes in the initial partial transaction, the user additionally spends an intent token note of value [1].
+The 'twin' intent token note of value [-1] balancing this token will only be created if the user's intent (intentVP) is satisfied.
+3. Once all of the parties involved in a state transition create their [-1] intent token notes (meaning that the intents of all of them are satisfied), the transaction is balanced and can be finalized.
+
+![img.png](img/exec_intent_notes.png)
+
+**Note**: instead of using notes of negative value the user can instead spend a [1] intent token note (that hasn't been created before) in their initial partial transaction. Once the user's intent is satisfied, a [1] intent note will be output, balancing the initial spend.
+
+**Note**: intent token notes are not committed to in the global CMtree
```

Comment:
As Joe [proposed](https://github.com/anoma/taiga/pull/85#issuecomment-1302925442), the intent notes will be committed to the tree. The code has been updated.

It is my understanding that intent notes(or general dummy notes) have no difference with "normal" notes except that intent notes don't need to check the merkle path authorization. Therefore, an intent note will be created as output-note(stored on the tree) in ActionCircuit when a user wants to create an intent token note of value [1]; then the intent note will be consumed as spent-note in ActionCircuit when the intent vp is satisfied and we can create the [-1] intent note. Finally, the sum of output-notes is balanced with the sum of spent-notes. Does it make sense?

Reply:
Ah, I didn't see this comment, thanks! Do we actually have negative value notes or we gonna have inverse life cycle (instead of creating [-1] and [1]  note, spend a [1] note first and then output it when the VP is happy)?

Reply:
In my current implementation, output an intent note first and spend it when the vp is satisfied.
We can have a discuss to see which is better?

Reply:
Not sure I understand the difference between outputting first and spending first (i'll drop a message on slack). I'll correct the spec to be in sync with the code